File: ii-service/src/main/java/com/manymonkeys/service/cinema/PersonService.java
package com.manymonkeys.service.cinema;

import com.manymonkeys.model.cinema.Person;
import com.manymonkeys.model.cinema.Role;
import com.manymonkeys.service.exception.NotFoundException;

/**
 * @author Ilya Pimenov
 *         Owls Proprietary
 */
public interface PersonService {

    Person createPerson(Person person);

    Person addRole(Person person, Role role) throws NotFoundException;

    Person findOrCreate(Person person);

    Person findOrCreate(Person person, Role role);
}

File: ii-service/src/main/scala/com/manymonkeys/service/cinema/PersonService.scala
package com.manymonkeys.service.cinema

import com.manymonkeys.model.cinema.Person
import com.manymonkeys.model.cinema.Role

/**
* @author Ilya Pimenov
*         Owls Proprietary
*/
trait PersonService {

  def createPerson(person: Person): Person

  def addRole(person: Person, role: Role): Person

  def findOrCreate(person: Person): Person

  def findOrCreate(person: Person, role: Role): Person

}

